synapse/storage/databases/main/event_push_actions.py

```diff
@@ -910,6 +910,8 @@ def _handle_new_receipts_for_notifs_txn(self, txn: LoggingTransaction) -> bool:
         # recalculate the summary.
         for _, room_id, user_id, stream_ordering in rows:
             # Only handle our own read receipts.
+            #
+            # XXX Is this needed, the query filters these out?
```

Comment:
This is the "recheck" referred to on line 895

I think the idea is to be wary of entries in `receipts_linearized` where `user_id` is something like `@user:evil.server:matrix.org`, which would match the query but not `is_mine_id`. Whether we *need* that sort of paranoia, I can't say.

Reply:
Ahhh, yeah that makes sense. 👍 I can remove this comment. I suspect doing it in the query was a first-level "let's return less data" while this is the actual check.